app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -17,7 +17,23 @@ def attempt(submission)
                          map { |answer| [answer.question, answer] }.to_h
 
     map do |question|
-      attempting_answers.fetch(question) { question.attempt(submission) }
+      attempting_answers[question] || question.latest_correct_answer(submission) ||
+        question.attempt(submission)
     end
   end
+
+  # Return the question at the given step.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which contains the answers.
+  # @return [Array<Course::Assessment::Question>] The question at the given the step. The latest
+  #   unfinished question will be returned if the question at the step is not accessible.
+  def step(submission, current_step)
+    # Make sure index is between [0, length - 1]
+    index = [length - 1, [0, current_step.to_i - 1].max].min
+
+    index -= 1 while index > 0 && !fetch(index - 1).finished_by?(submission)
```

Comment:
I *think* it can, finished is just a join to a submission, where the status is submitted or graded?